src/Microsoft.Azure.SignalR.Common/ClientInvocation/ICallerClientResultsManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal interface ICallerClientResultsManager
+    {
+        string GenerateInvocationId(string connectionId);
+
+        /// <summary>
+        /// Add a invocation which is directly called by current server
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="connectionId"></param>
+        /// <param name="invocationId"></param>
+        /// <param name="instanceId"> The InstanceId of target client the caller server knows when this method is called. If the target client is managed by the caller server, the caller server knows the InstanceId of target client and this parameter is not null. Otherwise, this parameter is null. </param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<T> AddInvocation<T>(string connectionId, string invocationId, string instanceId, CancellationToken cancellationToken);
+
+        void AddServiceMappingMessage(ServiceMappingMessage serviceMappingMessage);
```

Comment:
- Though this method uses `Dictionary.AddOrUpdate` , it's only designed to add a `(key, value)` rather than update one according to our business logic. 
- `Dictionary.AddOrUpdate` is used to avoid manual checking whether a `key` exists in the dictionary or not. It doesn't update the `value` corresponding to an existing `key`. 
```csharp
_serviceMappingMessages.AddOrUpdate(
    serviceMappingMessage.InstanceId,
    new List<string>() { serviceMappingMessage.InvocationId },
    (key, valueList) => { valueList.Add(serviceMappingMessage.InvocationId); return valueList; });
```
- So I guess the old naming is better
- They are not pair or symmetric.  `CleanupInvocations` cleans `_serviceMappingMessages` only works when a service is offline. `_serviceMappingMessages` can be cleaned when an invocation is completed normally as well.